tests/sentry/api/endpoints/test_group_similar_issues_embeddings.py

```diff
@@ -0,0 +1,157 @@
+                        "stacktrace": {
+                            "frames": [
+                                {
+                                    "function": "divide_by_zero_another",
+                                    "module": "__main__",
+                                    "filename": "python_onboarding.py",
+                                    "abs_path": "/Users/jodi/python_onboarding/python_onboarding.py",
+                                    "lineno": 20,
+                                    "context_line": " divide_by_zero_another()",
+                                }
+                            ]
+                        },
+                        "type": "ZeroDivisionError",
+                        "value": "division by zero",
+                    }
+                ]
+            },
+            "platform": "python",
+        }
+        self.event = self.store_event(data=self.base_error_trace, project_id=self.project)
+        self.group = self.event.group
+        assert self.group
+        self.path = f"/api/0/issues/{self.group.id}/similar-issues-embeddings/"
+
+    def test_get_stacktrace_string(self):
+        stacktrace_string = get_stacktrace_string(self.base_error_trace["exception"], self.event)  # type: ignore
+        assert stacktrace_string == EXPECTED_STACKTRACE_STRING
+
+    def test_get_stacktrace_string_no_values(self):
+        stacktrace_string = get_stacktrace_string({"values": []}, self.event)
+        assert stacktrace_string == ""
+
+    def test_no_feature_flag(self):
+        response = self.client.post(self.path)
+
+        assert response.status_code == 404, response.content
+
+    @with_feature("organizations:issues-similarity-embeddings")
+    @mock.patch(
+        "sentry.api.endpoints.group_similar_issues_embeddings.get_similar_issues_embeddings"
+    )
+    def test_simple(self, mock_get_similar_issues_embeddings):
+        similar_data = {
+            "responses": [
+                {
+                    "message_similarity": 0.95,
+                    "parent_group_id": 6,
+                    "should_group": True,
+                    "stacktrace_similarity": 0.99,
+                }
+            ]
+        }
+        mock_get_similar_issues_embeddings.return_value = similar_data
+        response = self.client.post(
+            self.path,
+            data={
+                "query": {"k": 1, "threshold": 0.98},
+            },
+        )
+        assert response.data == similar_data
+        mock_get_similar_issues_embeddings.assert_called_with(
+            {
+                "group_id": self.group.id,
+                "stacktrace": EXPECTED_STACKTRACE_STRING,
+                "message": self.group.message,
+                "k": 1,
+                "threshold": 0.98,
+            }
+        )
+
+    @with_feature("organizations:issues-similarity-embeddings")
+    @mock.patch(
+        "sentry.api.endpoints.group_similar_issues_embeddings.get_similar_issues_embeddings"
+    )
```

Comment:
Rather than mock this api call, I'd suggest we figure out how to use responses to mock the response here. Just make sure we have as close to an end-to-end test as possible

Reply:
good call, thanks you!